contracts/compliance/modular/modules/InitialLockupPeriodModule.sol

```diff
@@ -0,0 +1,160 @@
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *     This specific smart contract is also licensed under the Creative Commons
+ *     Attribution-NonCommercial 4.0 International License (CC-BY-NC-4.0),
+ *     which prohibits commercial use. For commercial inquiries, please contact
+ *     Tokeny sàrl for licensing options.
+ */
+
+pragma solidity 0.8.27;
+
+import "./AbstractModuleUpgradeable.sol";
+
+error InsufficientBalanceTokensLocked(address user, uint256 value, uint256 availableAmount);
+
+event LockupPeriodSet(address indexed compliance, uint256 lockupPeriod);
+
+
+contract InitialLockupPeriodModule is AbstractModuleUpgradeable {
+
+    struct LockedTokens {
+        uint256 amount;
+        uint256 releaseTimestamp;
+    }
+
+    mapping(address compliance => uint256 lockupPeriod) private _lockupPeriods;
+    mapping(address compliance => mapping(address user => LockedTokens[] lockedTokens)) private _lockedTokens;
+
+    /// @dev initializes the contract and sets the initial state.
+    function initialize() external initializer {
+        __AbstractModule_init();
+    }
+
+    /// @dev sets the lockup period for a compliance contract.
+    /// @param _lockupPeriod the lockup period in seconds.
+    function setLockupPeriod(uint256 _lockupPeriod) external onlyComplianceCall {
+        _lockupPeriods[msg.sender] = _lockupPeriod;
+
+        emit LockupPeriodSet(msg.sender, _lockupPeriod);
+    }
+
+    /// @inheritdoc IModule
+    function moduleTransferAction(address _from, address /*_to*/, uint256 _value) external override onlyComplianceCall { 
+        if (_from != address(0)) {
+            // Check if the user has enough unlocked tokens to transfer
+            uint256 availableAmount = _calculateAvailableAmount(msg.sender, _from);
+            if (_value > availableAmount) {
+                revert InsufficientBalanceTokensLocked(_from, _value, availableAmount);
+            }
+        }
+    }
+
+    /// @inheritdoc IModule
+    function moduleMintAction(address _to, uint256 _value) external override onlyComplianceCall {
+        _lockedTokens[msg.sender][_to].push(
+            LockedTokens({
+                amount: _value,
+                releaseTimestamp: block.timestamp + _lockupPeriods[msg.sender]
+            })
+        );
+    }
+
+    /// @inheritdoc IModule
+    // solhint-disable-next-line no-empty-blocks
+    function moduleBurnAction(address _from, uint256 _value) external override onlyComplianceCall { }
```

Comment:
need to manage the burning of locked tokens, otherwise you will have data on this module saying you have, for example 100 tokens unlocking in 3 days and 200 unlocking in 1 month while you only have 50 tokens left on your wallet. 
My opinion is that we need to burn first from the freely transferrable balance, and then from the balances that will be free the soonest. keeping the tokens that are going to unlock the latest untouched